core/src/rules/optimising_line_formatter/debug.rs

```diff
@@ -21,40 +21,22 @@ impl Debug for RawDebugLine<'_, '_, '_> {
         let minmax = self.line.1.get_tokens().iter().cloned().minmax();
         let (first, mut last) = minmax.into_option().unwrap_or((0, 0));
 
-        let get_next_line_parent = |line_children: &LineChildren| {
-            let last_child_line_index = *line_children.line_indices.last()?;
-            let last_child_line_token_index = *self
-                .iolf
-                .lines
-                .get(last_child_line_index)?
-                .get_tokens()
-                .last()?;
-            Some(LineParent {
-                line_index: last_child_line_index,
-                global_token_index: last_child_line_token_index,
-            })
-        };
-
-        let mut line_parent = LineParent {
+        let mut current_parent = LineParent {
             line_index: self.line.0,
-            global_token_index: *self
-                .line
-                .1
-                .get_tokens()
-                .iter()
-                .rev()
-                .find(|&&t| !self.iolf.token_types[t].is_comment_or_directive())
-                .unwrap_or(&last),
+            global_token_index: last,
```

Comment:
Are you sure this is correct in general? It could be a token in the middle of the line that is the parent. That's the case for the new change that was merged in #262.